src/lib/bit/bit.h

```diff
@@ -303,6 +303,220 @@ bit_set_range(void *data, size_t pos, size_t count, bool val)
 	 */
 }
 
+/**
+ * @brief Copy \a count bits from memory chunk \a src starting from bit \a src_i
+ *  into \a dst starting from bit \a dst_i.
+ * @param dst - the memory chunk to copy bits into.
+ * @param dst_i - the position to copy bits into.
+ * @param count - the amount of bits to copy.
+ * @param src - the memory chunk to copy bits from.
+ * @param src_i - the position to copy bits from.
+ */
+static inline void
+bit_copy_range(uint8_t *restrict dst, size_t dst_i, size_t count,
```

Comment:
What if `src` and `dst` point to overlapping memory, but source and destination chunks are not overlapping because of `src_i` and `dst_i`? It sounds legal. I mean:
```
[ ... src ... dst ... src[src_i] ... src[src_i+count] ... dst[dst_i] ... dst[dst_i+count] ... ]
```
In such case the `restrict` keywords could cause UB, so we should either forbid this or remove `restrict`.

Reply:
Yes, this function is to be used on scanning (copying data from leaf mask into a buffer) and batched insertion (copying data from the batch validity to leaf masks).

It can't be used for rebalancing (moving bits within a single leaf mask) even if we allow that, so I suggest to restrict that for the sake of performance.

Added an assertion for that:

```diff
--- a/src/lib/bit/bit.h
+++ b/src/lib/bit/bit.h
@@ -316,6 +316,8 @@ static inline void
 bit_copy_range(uint8_t *restrict dst, size_t dst_i, size_t count,
               const uint8_t *restrict src, size_t src_i)
 {
+       assert(dst != src);
+
        if (count == 0)
                return;
 
```

Reply:
> I suggest to restrict that for the sake of performance.

Please add a comment to the function description.

> `assert(dst != src);`

A more precise check would look like:
```c
assert(src > dst + DIV_UP(dst_i + count, CHAR_BIT) || dst > src + DIV_UP(src_i + count, CHAR_BIT));
```